src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/WcfService.cs

```diff
@@ -234,5 +236,46 @@ public XmlCompositeType GetDataUsingXmlSerializer(XmlCompositeType composite)
             composite.StringValue = composite.StringValue;
             return composite;
         }
+
+        public Stream GetStream(string data)
+        {
+            return StringToStream(data);
+        }
+
+        public bool UploadStream(Stream stream)
+        {
+            string data = StreamToString(stream);
+
+            if (!string.IsNullOrWhiteSpace(data))
+            {
+                return true;
```

Comment:
@iamjasonp raises a great point.  Regardless of the kind of data being sent or returned, we want to test each permutation of TransferMode is being used properly.  The case where Stream "auto chooses" a transfer mode is a special case (I think). After this is checked in, we should articulate which scenarios require a server endpoint that itself sets TransferMode.  I think we may need a new HttpResource devoted to the different TransferModes (and verify simple and complex types stream both ways properly.  I think we will also need TCP flavors of all the streamed test, but let's make that a new issue and not tackle it here.